ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -367,23 +367,58 @@ func (r *RayJobReconciler) getOrCreateRayClusterInstance(ctx context.Context, ra
 	rayClusterInstance := &rayv1alpha1.RayCluster{}
 	err := r.Get(ctx, rayClusterNamespacedName, rayClusterInstance)
 	if err == nil {
-		r.Log.Info("RayJob associated rayCluster found", "rayjob", rayJobInstance.Name, "raycluster", rayClusterNamespacedName)
-		// just return the rayClusterInstance if cluster spec is nil
+		r.Log.Info("Found associated RayCluster for RayJob", "rayjob", rayJobInstance.Name, "raycluster", rayClusterNamespacedName)
+
+		// Case1: The job is submitted to an existing ray cluster, simply return the rayClusterInstance.
 		if rayJobInstance.Spec.RayClusterSpec == nil {
```

Comment:
With the current logic:

1. `RayClusterSpec` empty, `clusterSelector` set: Existing RayCluster is used.
2. `RayClusterSpec` empty, `clusterSelector` unset: Triggers an error for RayCluster creation.
3. `RayClusterSpec` set, `clusterSelector` set: Existing RayCluster is used, RayClusterSpec is ignored. However, this could result in a conflict as both the RayJob controller and the autoscaler in the existing RayCluster might try to update replicas simultaneously.
4. `RayClusterSpec` set, `clusterSelector` unset: A new RayCluster is created based on `RayClusterSpec`.

In summary, if `clusterSelector` is set, `RayClusterSpec` is ignored under the current logic. But this might result in scenario 3.

To address scenario 3 and simplifies the logic, How about change the logic to be::

1. `clusterSelector` set: The existing RayCluster is used. Modifications to the existing RayCluster by the RayJob controller would not be permitted (this would avoid scenario 3).
2. `clusterSelector` unset: A new RayCluster is created based on `RayClusterSpec` (obviously, this will trigger an error for RayCluster creation if `RayClusterSpec` is nil).

This change maintains backward compatibility.